datafusion/physical-expr/src/math_expressions.rs

```diff
@@ -701,6 +701,29 @@ macro_rules! make_try_abs_function {
     }};
 }
 
+// Decimal types should keep the same precision and scale by using `with_data_type()`.
+// https://github.com/apache/arrow-rs/issues/4644
+macro_rules! make_decimal_abs_function {
+    ($ARRAY_TYPE:ident) => {{
+        |args: &[ArrayRef]| {
+            let array = downcast_arg!(&args[0], "abs arg", $ARRAY_TYPE);
+            let res: $ARRAY_TYPE = array
+                .try_unary(|x| {
+                    // Valid Decimal values should never overflow
+                    x.checked_abs().ok_or_else(|| {
```

Comment:
Could you add a test for overflow case?

Reply:
@viirya The `abs` function only panics on `MIN`, which is not a valid decimal number unless there is a bug somewhere.

Reply:
If so, this `checked_abs` looks unnecessary. Maybe just `wrapping_abs`?

Reply:
That's said, if you assume that input decimal array could be "invalid", we will have this kind of check in every operator/expressions. 😄

Reply:
@viirya 
Not all operators/expressions require this. Overflow in `abs` here is a bug, while others like `sum` may overflow even if inputs are valid.

I have considered using `wrapping_abs`, but I’m worried that if invalid values are introduced elsewhere, they might be hidden and not easily detected. 

It seems to be a trade-off between performance and safety, but for me, both are ok.

I will follow the suggestion and change it to `wrapping_abs`.

Reply:
It is not a bug...If you feed invalid data, the operators/expressions won't perform correctly.

I mean, if you worry about `abs`'s input could be invalid decimal values, you will worry other operator/expression takes invalid input too. It doesn't make sense.

If you have a decimal input with invalid values (for example, it is out of the representation range of the precision/scale), some operators/expressions will perform wrongly. But it is not a reason to add such check to all operators/expressions to verify if input is valid or not. It means that you will have to validate inputs of each operator/expression.

I think `wrapping_abs` is enough here.

Reply:
Thank you @viirya, it makes sense to me.